data_structures/src/chain.rs

```diff
@@ -1877,8 +1923,53 @@ impl TransactionsPool {
         (reveals_vector, total_fee)
     }
 
+    /// Remove transactions until the size limit is satisfied.
+    /// Returns a list of all the removed transactions.
+    fn remove_transactions_for_size_limit(&mut self) -> Vec<Transaction> {
+        let mut removed_transactions = vec![];
+
+        while self.total_vt_size_bytes + self.total_dr_size_bytes > self.weight_limit {
+            // Try to split the memory between value transfer and data requests using the same
+            // ratio as the one used in blocks
+            // The ratio of vt to dr in blocks is currently 4:1
+            // TODO: calculate this from ConsensusConstants?
```

Comment:
Please!

Reply:
Done, but need tests to ensure that a ratio of 4:1 is 4.0 and not 0.25